src/components/exercise/index.cjsx

```diff
@@ -122,16 +120,14 @@ ExerciseWithScroll = React.createClass
 
   componentDidMount: ->
     {currentStep} = @props
-    if currentStep?
-      @scrollToSelector("[data-step='#{currentStep}']")
+    @scrollToStep(currentStep) if currentStep?
 
   componentWillReceiveProps: (nextProps) ->
-    if nextProps.currentStep isnt @props.currentStep
-      @scrollToSelector("[data-step='#{nextProps.currentStep}']")
+    @scrollToStep(nextProps.currentStep) if nextProps.currentStep isnt @props.currentStep
 
-  onAfterScroll: ->
-    textArea = @_scrollingTargetDOM().querySelector("[data-step='#{@props.currentStep}'] textarea")
-    textArea?.focus()
+  scrollToStep: (currentStep) ->
+    stepSelector = "[data-step='#{currentStep}']"
+    @scrollToSelector(stepSelector, updateHistory: false) unless @isSelectorInView(stepSelector)
 
```

Comment:
hmm...i think this is more clear than adding an option, but i guess this would likely be common enough to where we wouldnt be wanting to repeat this all the time